### ISSUE TITLE
Proposal - XML Comments on program entry point

### ISSUE THREAD

**Issue by Foxtrek64** [DotNetAnalyzers/StyleCopAnalyzers#3479]
Currently if you have a simple program laid out as follows:

```cs
public class Program
{
    public static void Main()
    {
        Console.WriteLine("Hello, World!");
    }
}
```

StyleCop will correctly flag both `Program` and `Main()` for CS1591 (Missing XML Comment) and SA1600 (Element must be documented). I have gotten in the habit of suppressing both of these issues at the top of the file since it's my opinion that these elements do not need to be documented. XML comments, in my mind, serve to inform contributors and consumers of a project about the purpose of the class and function. Contributors need no introduction to the function of either `Program` or `Main()` and this class is not present in a class library, leading to no utility for consumers.

I have attempted suppressing these in my .editorconfig by adding this configuration section to the bottom:
```ini
[Program.cs]
# Documenting Program and Main() is not necessary.
dotnet_diagnostic.CS1591.severity = none
dotnet_diagnostic.SA1600.severity = none
```
However, this appears to only suppress SA1600, leaving CS1591 untouched.

I would like to propose one of the following solutions:

1. Suppress these two compiler warnings specifically for the program entry point. Potential drawbacks to this being I'm unsure how granular we can make this. We will want to avoid suppressing this issue for public members of Program other than `Main()`.
2. Include default documentation for `Program` and `Main()` similarly to what is included for constructors. Some considerations for this include perhaps allowing the customization of this default documentation text in case of any domain-specific standards and not raising SA1642 (Documentation must begin with standard text) in the event that it doesn't match.

I welcome thoughts and concerns on this. 

Edit: A couple of findings.

Removing the explicit accessibility modifiers will remove the CS1591 error. However, this may in turn raise SA1400 (Element should declare an accessibility modifier). SA1600 remains unaffected, still requiring suppression.

Removing the namespace, class, and Main() declarations (default template in .NET 6) resolves the issue by simply removing any elements that need documenting in the first place. I don't consider this to be too ideal though as it prevents you from declaring class members of `Program`, which may be useful under some circumstances.

**Comment by sharwell** [comment]
> leaving CS1591 untouched

CS1591 is reported by the C# compiler, so this project has no control over it.

> Suppress these two compiler warnings specifically for the program entry point.

The program entry point does not need to be public. My recommendation is to resolve this by the following change:

```patch
-public class Program
+internal class Program
 {
     public static void Main()
     {
         Console.WriteLine("Hello, World!");
     }
 }
```

**Comment by Foxtrek64** [comment]
This combined with the change to .editorconfig to set the severity of SA1600 to none removes all errors related to CS1591 and SA1600.
